### ISSUE TITLE
Any idea where this strange output/formatting is coming from?

### ISSUE THREAD

**Issue by iokiwi** [tgenv/tgenv#10]
MacOS Ventura 13.0.1
M1 Macbook Air

Also reproducible in docker with ubuntu:latest.

```bash
docker run -it ubuntu:latest
apt update -y
apt install git curl
git clone https://github.com/tgenv/tgenv.git ~/.tgenv
echo 'export PATH="$HOME/.tgenv/bin:$PATH"' >> ~/.bash_profile
source ~/.bashrc
tgenv install latest
```

Not a big issue, just a weird nit pick.

![Screenshot 2023-01-11 at 11 02 46](https://user-images.githubusercontent.com/1480457/211671946-b607e3ed-73d6-4718-be9d-545dc75c0022.png)

**Comment by iokiwi** [comment]
It's to do with the progress bar flag (`-#`) given to curl. I can reproduce the output just by running the following cURL based on what [tgenv-install does](https://github.com/tgenv/tgenv/blob/main/libexec/tgenv-install#L77).

```bash
curl -# --tlsv1.2 --location-trusted -f -L -o "./terragrunt" "https://github.com/gruntwork-io/terragrunt/releases/download/v0.42.4/terragrunt_darwin_arm64"
```

So, the output is coming from curl.

I wondered why `tfenv` doesn't exhibit the same problem output. As far as I can see, our curl request is very similar to the one used in tfenv.

The main difference appears to be that `tfenv` installs from `releases.hashicorp.com` where as tgenv is installing from github where `https://github.com/gruntwork-io/terragrunt/releases/download/v0.42.7/terragrunt_darwin_amd64` ultimately resolves to `https://objects.githubusercontent.com/github-production-release-asset-2e65be/59522149/4d92e21e-8f28-4615-9a7e-0d5ac7ecf810[...]`.

I get similar output downloading terraform source zip from github as well. 

```bash
curl -# --tlsv1.2 --location-trusted -f -L -o "./terragrunt" https://github.com/hashicorp/terraform/archive/refs/tags/v1.3.7.zip
```

![Screenshot 2023-01-11 at 12 15 35](https://user-images.githubusercontent.com/1480457/211682136-18c8565a-43e6-413c-99b6-b7e24df8c2c7.png)


## Conclusion and suggestion for aesthetic improvement

It appears the output is an artifact of the download endpoint (github's server vs hashicorp's server).

Given that the output is a result of downloading from github which is out of our hands, I suggest we either change our curl statement to just be silent (`-s`) rather than use the progress bar (`-#`).

**Comment by weyderfs** [comment]
@iokiwi Are you using `sh` such a shell right?

I just reproduced here using `bash` and the output have been showed fine in the _screen_:

`$ docker container run -it ubuntu:latest bash`

```bash
root@1e7a184a539b:~# tgenv install latest
[INFO] Installing Terragrunt v0.42.7
[INFO] Downloading release tarball from https://github.com/gruntwork-io/terragrunt/releases/download/v0.42.7/terragrunt_linux_amd64
########################################################################################################################################################################### 100.0%
[INFO] Installation of terragrunt v0.42.7 successful
[INFO] Switching to v0.42.7
[INFO] Switching completed
root@1e7a184a539b:~# history
    1  apt update -y
    2  apt install git curl
    3  git clone https://github.com/tgenv/tgenv.git ~/.tgenv
    4  echo 'export PATH="$HOME/.tgenv/bin:$PATH"' >> ~/.bash_profile
    5  ls -la
    6  cd /root/
    7  ls -la
    8  source .bashrc
    9  pwd
   10  source .bash_profile
   11  tgenv install latest
   12  history
```
I'm using a Macbook Pro `amd64`
```sh

❯ neofetch                                                                                                                                                                     ─╯
                    'c.          
                 ,xNMM.          -----------------------------------
               .OMMMMo           OS: macOS 13.1 22C65 x86_64
               OMMM0,            Host: MacBookPro16,2
     .;loddo:' loolloddol;.      Kernel: 22.2.0
   cKMMMMMMMMMMNWMMMMMMMMMM0:    Uptime: 3 days, 3 hours, 35 mins
 .KMMMMMMMMMMMMMMMMMMMMMMMWd.    Packages: 143 (brew)
 XMMMMMMMMMMMMMMMMMMMMMMMX.      Shell: zsh 5.8.1
;MMMMMMMMMMMMMMMMMMMMMMMM:       Resolution: 1440x900@2x
:MMMMMMMMMMMMMMMMMMMMMMMM:       DE: Aqua
.MMMMMMMMMMMMMMMMMMMMMMMMX.      WM: Quartz Compositor
 kMMMMMMMMMMMMMMMMMMMMMMMMWd.    WM Theme: Blue (Light)
 .XMMMMMMMMMMMMMMMMMMMMMMMMMMk   Terminal: iTerm2
  .XMMMMMMMMMMMMMMMMMMMMMMMMK.   Terminal Font: MesloLGS-NF-Regular 13
    kMMMMMMMMMMMMMMMMMMMMMMd     CPU: Intel i5-1038NG7 (8) @ 2.00GHz
     ;KMMMMMMMWXXWMMMMMMMk.      GPU: Intel Iris Plus Graphics
       .cooc,.    .,coo:.        Memory: 10514MiB / 16384MiB

```

I could say the glitch it's about the `curl` version, but we are using the same version `ubuntu:latest`, I'm curious now hahahaha

**Comment by iokiwi** [comment]
Weird I don't see it on all machines. 

Added context: On my mac im using iterm2 with zsh.

zsh shouldn't matter because the script has a shebang that makes it run under bash. I've checked and the bash version similar in macos and ubuntu.

I see it if my terminal is maximized but not always when my terminal is small/windowed.

TBH I think I will just close this. Its not a big deal and, as described, its coming from curl. I also kind of like the progress bar when its not bugging out.
